nvflare/fuel/utils/config_factory.py

```diff
@@ -106,3 +108,25 @@ def get_config_loader(config_format: ConfigFormat) -> Optional[ConfigLoader]:
         if config_format is None:
             return None
         return ConfigFactory._fmt2Loader.get(config_format)
+
+    @staticmethod
+    def match_config(parent, init_file_path, match_fn) -> bool:
+        # we ignore the original extension
+        basename = os.path.splitext(pathlib.Path(init_file_path).name)[0]
+        ext2fmt_map = ConfigFormat.config_ext_formats()
+        for ext in ext2fmt_map:
+            if match_fn(parent, f"{basename}{ext}"):
+                return True
+        return False
+
+    @staticmethod
+    def has_config(init_file_path: str, search_dirs: Optional[List[str]] = None) -> bool:
+        fmt, real_file_path = ConfigFactory.search_config_format(init_file_path, search_dirs)
+        return real_file_path is not None
+
+    @staticmethod
+    def _os_walk(search_dir):
```

Comment:
You can use mock in the unit test to avoid wrap this into a separate function only for the unit test purpose.

Reply:
This is how to do it:

    def test_config_search(self, init_file_path, search_dirs, expected_loc, expected_fmt):

        os.walk = MagicMock(side_effect=self._walk_result)
        # ConfigFactory._os_walk = self._walk_result
        fmt, location = ConfigFactory.search_config_format(init_file_path, search_dirs)
        assert fmt == expected_fmt
        assert location == expected_loc